FWCore/Framework/src/StreamSchedule.cc

```diff
@@ -539,17 +535,35 @@ namespace edm {
   
   void StreamSchedule::processOneEventAsync(WaitingTaskHolder iTask,
                                             EventPrincipal& ep,
-                                            EventSetup const& es)
-  {
+                                            EventSetup const& es,
+                                            std::vector<edm::propagate_const<std::shared_ptr<PathStatusInserter>>>& pathStatusInserters) {
```

Comment:
Maybe it would be better to determine the empty path inserters up front and just keep that list? That way there are no filtering to do during this call.